js/pathfinding.js

```diff
@@ -5,13 +5,59 @@ import {settingsKey} from "./settings.js";
 import {getSnapPointForTokenObj, iterPairs} from "./util.js";
 
 import * as GridlessPathfinding from "../wasm/gridless_pathfinding.js";
-import {MaxSizeStackSet, PriorityQueueSet} from "./data_structures.js";
+import {PriorityQueueSet} from "./data_structures.js";
 
-const maxCacheIds = 5;
-const cache = {
-	nodes: new Map(),
-	cacheIds: new MaxSizeStackSet(maxCacheIds)
+class Cache {
+	constructor() {
+		this.nodes = new Map();
+		this.lastUsed = new Map();
+	}
+
+	clear() {
+		this.nodes.clear();
+		this.lastUsed.clear();
+	}
+
+	/**
+	 * Get the cache associated with the given cache ID, creating a new one
+	 * if we don't already have one
+	 */
+	getCachedNodes(cacheId) {
+		// Track that we've last used this cache right now
+		this.lastUsed.set(cacheId, Date.now());
+
+		// Get the nodes for the cacheId. If we don't already have one, create one
+		let cachedNodes = this.nodes.get(cacheId);
+		if (!cachedNodes) {
+			cachedNodes = new Array(gridHeight);
+			for (let y = 0; y < gridHeight; y++) {
+				cachedNodes[y] = new Array(gridWidth);
+				for (let x = 0; x < gridWidth; x++) {
+					cachedNodes[y][x] = {x, y};
+				}
+			}
+			this.nodes.set(cacheId, cachedNodes);
+
+			// Since we're adding a new cache, check if we have too many and,
+			// if we do, get rid of the one that was last used longest ago
+			if (this.lastUsed.size > maxCacheIds) {
```

Comment:
I think having `maxCacheId` as an instance variable of the `Cache` class is preferable to using a global Variable